_posts/2020-10-13-intro-to-hive-connector.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: post
+title:  "A gentle introduction to the Hive connector"
+author: Brian Olsen
+excerpt_separator: <!--more-->
+---
+One of the most confusing aspects when starting Presto is the Hive connector.
+
+Typically, you seek out the use of Presto when you experience an intensely slow
+query turnaround from your existing Hadoop or Hive infrastructure. In fact, the
+genesis of Presto came about due to these slow Hive query conditions at Facebook
+back in 2012. So when you learn that Presto has a Hive connector, it can be
+rather confusing since you moved to Presto to circumvent the slowness of your
+current Hive cluster. Another common source of confusion is when you want to 
+query your data from your S3 compatible object storage, such as AWS S3, MinIO, 
+and Google Cloud Storage. This too uses the Hive connector. If that confuses
+you, don't worry, you are not alone. This blog aims to explain this commonly
+confusing nomenclature.
+
+<!--more-->
+
+## Hive Architecture
+
+To understand the origins and inner workings of Presto's Hive connector, you
+first need to know a few high level components of the Hive architecture.
+
+![](/assets/blog/intro-to-hive-connector/hive.png)
+
+You can simplify the Hive architecture to three elements;
+_the runtime_, _the metastore_, and _the storage_. _The runtime_ contains the
+logic of the query engine that translates the SQL-esque Hive Query Language
+(HQL) into MapReduce jobs that run over files stored in the filesystem.
+_The storage_ component is simply that, it stores files in various formats and
```

Comment:
Separate paragraphs for the three components .. esp since the storage one is long..